core/src/main/scala/tapir/Endpoint.scala

```diff
@@ -116,9 +116,67 @@ case class Endpoint[I, E, O, +S](input: EndpointInput[I], errorOutput: EndpointO
     */
   def showRaw: String = toString
 
+  /**
+    * Renders endpoint path, by default all parametrised path and query components are replaced by {param_name} or {paramN}, e.g. for
+    * {{{
+    * endpoint.in("p1" / path[String] / query[String]("par2"))
+    * }}}
+    * returns `/p1/{param1}?par2={par2}`
+    */
+  def renderPath(
```

Comment:
Two suggestions:
1. maybe this could be called `renderPathTemplate` or sth like this? As this is essentialy a template
2. maybe we could move the code to a separate object? It could then also contain the defaults

Reply:
1. sure ;)
2. what do you exactly mean? something similar to `EndpointToOpenAPIDocs` with implicit classes for `Endpoint` so it is not by default built into Endpoint? If so, where should it be located? I suppose the same module, but by import `tapir.template._`/`tapir.path._`, something else?

Reply:
no, just an object with an implementation, with the code moved to a separate object. Built into Endpoint is fine :)